### PR TITLE
Task #54842 [LiconKA] Function-2 : Display rss by category

### PR DESCRIPTION
**Pull request by liconrepo**

### CODE REVIEW

File: app/src/main/java/com/licon/rssfeeds/data/constants/AppData.java [hunk 1]
```diff
@@ -4,10 +4,11 @@
  * Created by FRAMGIA\khairul.alam.licon on 24/2/16.
  */
 public class AppData {
-	public static final int SPLASH_TIME_OUT = 2000;
-	public static final int TAB_TECHNOLOGY = 0;
-	public static final int TAB_BUSINESS = 1;
-	public static final int TAB_FINANCE = 2;
-	public static final int TAB_TRAVEL = 3;
-	public static final int TAB_OTHERS = 4;
+	public static final int SPLASH_TIME_OUT = 1000;
```
Comment (phamxuanlu) [inline app/src/main/java/com/licon/rssfeeds/data/constants/AppData.java]: Please config to use 4 space instead hard tab for code indent

Comment (phamxuanlu) [inline app/src/main/java/com/licon/rssfeeds/data/constants/AppData.java]: And reformat all of your code

File: app/src/main/java/com/licon/rssfeeds/ui/adapter/RssBaseAdapter.java [hunk 2]
```diff
@@ -18,7 +18,7 @@
 
 import java.util.List;
 
-public class TechnologyAdapter extends RecyclerView.Adapter<TechnologyAdapter.SimpleViewHolder> {
+public class RssBaseAdapter extends RecyclerView.Adapter<RssBaseAdapter.SimpleViewHolder> {
     private static final String PUBLISHED_ON = "Published on : ";
```
Comment (phamxuanlu) [inline app/src/main/java/com/licon/rssfeeds/ui/adapter/RssBaseAdapter.java]: Don't use hard string in your code like this. You can put it into string.xml with pattern and get it via context.getResources().getString(int StringId)
ex:
<string name="rss_publish_on">Publish on : %s"</string>

File: app/src/main/java/com/licon/rssfeeds/ui/fragment/RssBaseFragment.java [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+                mLayoutErrorLoad.setVisibility(View.GONE);
+            }
+        });
+    }
+
+    void addData(final FeedItem item) {
+        if (getActivity() == null)
+            return;
+
+        getActivity().runOnUiThread(new Runnable() {
```
Comment (phamxuanlu) [inline app/src/main/java/com/licon/rssfeeds/ui/fragment/RssBaseFragment.java]: Don't need use runOnUiThread. It's already run on Main Thread.

File: app/src/main/java/com/licon/rssfeeds/ui/activity/MainActivity.java [hunk 4]
```diff
@@ -19,111 +20,163 @@
...
+        mDrawerToggle.syncState();
+    }
+
+    private void setupViewPagerWithTabs() {
+        mViewPagerAdapter = new ViewPagerAdapter(getSupportFragmentManager());
+        mViewPager.setAdapter(mViewPagerAdapter);
+        mTablayout.setupWithViewPager(mViewPager);
+
+        //tab-0
+        final TabLayout.Tab tab_usa_home = mTablayout.newTab();
```
Comment (nvmanh) [inline app/src/main/java/com/licon/rssfeeds/ui/activity/MainActivity.java]: why not create method for these?
void addTab(int textResId, int iconResId){
          TabLayout.Tab tab = mTablayout.newTab();
          tab.setText(getResources().getString(R.string.rss_category_business));
          tab.setIcon(getResources().getDrawable(R.drawable.ic_tab_business));
          mTablayout.addTab(tab_business, AppData.TAB_BUSINESS);
}
And all code like these will be add in method above

File: app/src/main/java/com/licon/rssfeeds/ui/fragment/RssBaseFragment.java [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+            public void run() {
+                mFeedsData.add(item);
+                mRssBaseAdapter.notifyDataSetChanged();
+            }
+        });
+    }
+
+    private void loadData() {
+        showLoading();
+        new AsyncTask<Void, Integer, Reader>() {
```
Comment (nvmanh) [inline app/src/main/java/com/licon/rssfeeds/ui/fragment/RssBaseFragment.java]: please return string from doInBackground and using it in onPost(). Please reading asyntask android first

File: app/src/main/res/layout/fragment_others.xml [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+                        android:gravity="center"
+                        android:paddingTop="@dimen/grid_padding_top"
+                        android:text="@string/rss_category_africa"
+                        android:textColor="@color/color_gray_dark"
+                        android:textSize="@dimen/text_size_sp_22"/>
+
+                    <Button
+                        android:id="@+id/button_asia"
+                        android:layout_width="@dimen/grid_hw_for_weight"
+                        android:layout_height="fill_parent"
```
Comment (nvmanh) [inline app/src/main/res/layout/fragment_others.xml]: buttons or views hve same style like this, please create your style in styles.xml and add style to them. Please read how to define and use style in android

File: app/src/main/java/com/licon/rssfeeds/ui/adapter/RssBaseAdapter.java [hunk 7]
```diff
@@ -45,11 +45,12 @@ public void onBindViewHolder(final SimpleViewHolder holder, int position) {
 
         String title = feed.getTitle();
         String description = feed.getDescription();
-        final String published_date = PUBLISHED_ON + DateFormatUtil.parseDateToString(feed.getPublicationDate());
+        String published_on = String.format(mContext.getResources().getString(R.string.published_on),
+                DateFormatUtil.parseDateToString(feed.getPublicationDate()));
 
         holder.mTextTitle.setText(TextUtils.isEmpty(title) ? "" : title);
```
Comment (nvmanh) [inline app/src/main/java/com/licon/rssfeeds/ui/adapter/RssBaseAdapter.java]: please make function for these:
addTextToView(TextView, String text){
   if(TextUtils.isEmpty(text))text="";
   textView.setText(text);
}

File: app/src/main/java/com/licon/rssfeeds/ui/fragment/RssBaseFragment.java [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+                    public void OnFeedItem(FeedParser feedParser, FeedItem item) {
+                        feedItems.add(item);
+                    }
+                });
+                XmlPullParser parser = Xml.newPullParser();
+                parser.setInput(result);
+                feedParser.parseFeed(parser);
+
+            } catch (IOException e) {
+                showError();
```
Comment (nvmanh) [inline app/src/main/java/com/licon/rssfeeds/ui/fragment/RssBaseFragment.java]: if error please return null onPostExecute and handle error at that place not here

Comment (liconrepo) [inline app/src/main/java/com/licon/rssfeeds/ui/fragment/RssBaseFragment.java]: I still can't understand this issue.

File: app/src/main/java/com/licon/rssfeeds/ui/fragment/RssBaseFragment.java [hunk 9]
```diff
@@ -0,0 +1,165 @@
...
+                feedParser.parseFeed(parser);
+            } catch (IOException e) {
+                showErrorToReload();
+            } catch (XmlPullParserException xmlPullParserException) {
+                UIUtil.showErrorDialogNotify(getActivity(),
+                        getString(R.string.text_dialog_title_error),
+                        xmlPullParserException.getMessage(),
+                        getString(R.string.text_dialog_btn_ok));
+            } catch (FeedParser.UnknownFeedException unknownFeedException) {
+                UIUtil.showErrorDialogNotify(getActivity(),
```
Comment (nvmanh) [inline app/src/main/java/com/licon/rssfeeds/ui/fragment/RssBaseFragment.java]: return null.

File: app/src/main/java/com/licon/rssfeeds/ui/fragment/RssBaseFragment.java [hunk 10]
```diff
@@ -0,0 +1,165 @@
...
+                        feedItems.add(item);
+                    }
+                });
+                XmlPullParser parser = Xml.newPullParser();
+                parser.setInput(result);
+                feedParser.parseFeed(parser);
+            } catch (IOException e) {
+                showErrorToReload();
+            } catch (XmlPullParserException xmlPullParserException) {
+                UIUtil.showErrorDialogNotify(getActivity(),
```
Comment (nvmanh) [inline app/src/main/java/com/licon/rssfeeds/ui/fragment/RssBaseFragment.java]: return null

File: app/src/main/java/com/licon/rssfeeds/ui/fragment/RssBaseFragment.java [hunk 11]
```diff
@@ -0,0 +1,165 @@
...
+                    @Override
+                    public void OnFeedItem(FeedParser feedParser, FeedItem item) {
+                        feedItems.add(item);
+                    }
+                });
+                XmlPullParser parser = Xml.newPullParser();
+                parser.setInput(result);
+                feedParser.parseFeed(parser);
+            } catch (IOException e) {
+                showErrorToReload();
```
Comment (nvmanh) [inline app/src/main/java/com/licon/rssfeeds/ui/fragment/RssBaseFragment.java]: return null

**Comment by nvmanh** [comment]
OK!

**Pull request merged**